app/src/main/java/com/simplecity/amp_library/ui/screens/nowplaying/PlayerPresenter.kt

```diff
@@ -302,8 +303,8 @@ class PlayerPresenter @Inject constructor(
     }
 
     fun saveUiEvent(uiEventType: UiEventType){
-        if (PlayerFragment.getSong() != null) {
-            val uiEvent = EventUtils.newUiEvent(PlayerFragment.getSong(), uiEventType, context)
+        if (MusicServiceUtils.getSong() != null) {
+            val uiEvent = EventUtils.newUiEvent(MusicServiceUtils.getSong()!!, uiEventType, context)
```

Comment:
You can avoid the `!!` here by doing something like this:

```kotlin
Song song = MusicServiceUtils.getSong()
 if (song  != null) {
            val uiEvent = EventUtils.newUiEvent(song, uiEventType, context)
...
```

...because you'd guaranteeing that the `song` variable won't be `null` at that point. That's better than potentially having the state change between the two invocations of `MusicServiceUtils.getSong()` which could potentially end up throwing an NPE at the 2nd invocation.

Reply:
Sounds good. Thank you for letting me know. I'll make it similar for other `!!` in the code.